src/lib/util/TelemetryUtil.ts

```diff
@@ -0,0 +1,17 @@
+import { Lifecycle, Logger, Messages } from '@salesforce/core';
+import { TelemetryData } from '../../types';
+
+let LOGGER: Logger = null;
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages("@salesforce/sfdx-scanner", "TelemetryUtil");
+
+export async function emitTelemetry(data: TelemetryData): Promise<void> {
+	try {
+		await Lifecycle.getInstance().emitTelemetry(data);
+	} catch (e) {
+		// Log a warning about the failure to emit telemetry.
+		LOGGER = await Logger.child('TelemetryUtil');
```

Comment:
Looks like logger initialization will happen every time. This may need a `!isInitialized` or a null check.

Reply:
Good catch.